### PR TITLE
Change variable 't' to 'r'

### LINKED ISSUE #262: AuthenticationError
I've been facing this "AuthenticationError: Could not find variable t on Live login page", its started all of the sudden, I will do more testing with user agents. Just wondering if anyone else is getting this?

### PR DESCRIPTION
**Pull request by carterschieffer**
Fixes #262

### CODE REVIEW

**Comment by amayer5125** [comment]
I will test this when I get home.

**Comment by specu** [comment]
this worked for me. @zengu94 's fix, patch #264 (@kylef000) also worked for me (although that one doesnt look as pretty)

**Comment by zengfu94** [comment]
While this does work, long term generating the PPSX would prevent further issues.

**Comment by kenneyhe** [comment]
the authorization error line should be deleted.. I tested below and it is working for me

         PPFT = page[s:e]
 
-# get PPSX parameter
-        ppsxSearch = self.ppsxValue.search(page)
-        if ppsxSearch == None:
-            raise AuthenticationError("Could not find variable 't' on Live login page")
-        PPSX = ppsxSearch.group(1)
+        PPSXs = ["P","Pa","Pas","Pass","Passp","Passpo","Passpor","Passport","PassportR","PassportRN"]
+        PPSX = random.choice(PPSXs)

**Comment by carterschieffer** [comment]
@kenneyhe Not sure what you mean. You shouldn't have to delete anything. It looks like you have code from the other pull request.

**Comment by kenneyhe** [comment]
Per zengfu94 comment, patch #264 would conflict/#265 and un-necessary.  @carterschieffer , I have cleaned up the Facebook code in bingAuth.py on my side

**Comment by amayer5125** [comment]
@kenneyhe I am working on cleaning up the facebook auth also. I am going a small general overhaul.

File: pkg/bingAuth.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def handle_starttag(self, tag, attrs):
 class BingAuth:
     inputNameValue = re.compile(r"<input.+?name=\"(.+?)\".+?value=\"(.+?)\"")
     formAction = re.compile(r"<form.+action=\"(.+?)\"")
-    ppsxValue = re.compile(r",t:'(.+?)',")
+    ppsxValue = re.compile(r",r:'(.+?)',")
```
Comment (amayer5125) [inline pkg/bingAuth.py]: If this is either `t` or `r` maybe this should be `ppsxValue = re.compile(r",[tr]:'(.+?)',")`
If you make this change and increment the version and date in main.py I will merge this.

**Comment by amayer5125** [comment]
@zengfu94 how would we generate this value? isnt this something that is passed from MS?

**Comment by zengfu94** [comment]
@amayer5125 the login succeeds even with it blank, the value is a random combination of 'PassportRN', so it could be just 'P' or 'Passt' or 'Passport'. easy to generate in a one liner

**Comment by amayer5125** [comment]
Oh I thought that was just something random they came up with in the other PR. Is that somewhere on the page we can pull?

**Comment by zengfu94** [comment]
should be in either r or t but if not i say generate

**Comment by zengfu94** [comment]
like a t.search if none r.search if none generate

**Comment by amayer5125** [comment]
@zengfu94 I'll make something quick and make a PR. Would you be willing to give it a thumbs up/down?

**Comment by zengfu94** [comment]
sure

**Comment by amayer5125** [comment]
@zengfu94 what if we just look for `:'(Pa?s?s?p?o?r?t?R?N?)'`

Not sure how familiar you are with regex.

**Comment by zengfu94** [comment]
Not very familiar but could this match just P?

**Comment by zengfu94** [comment]
Id say that'll work then

**Comment by amayer5125** [comment]
Yes. I just tested it 25 times and one of them was just P. It matched fine. PR in coming.

Here is a handy tool for testing regex: https://regex101.com/r/ssgXRK/1  
(green means match)

**Comment by amayer5125** [comment]
Implemented in #266 

Thank you for your help!

**Pull request closed without merging**